golang/widget-store/main.go

```diff
@@ -70,16 +29,21 @@ func main() {
 		logger.Fatal("DBOS_SYSTEM_DATABASE_URL required")
 	}
 
-	err := dbos.Initialize(dbos.Config{AppName: "widget_store_go", DatabaseURL: os.Getenv("DBOS_SYSTEM_DATABASE_URL")})
+	dbosContext, err := dbos.NewDBOSContext(dbos.Config{
+		AppName:     "widget_store_go",
+		DatabaseURL: os.Getenv("DBOS_SYSTEM_DATABASE_URL"),
+	})
 	if err != nil {
 		logger.WithError(err).Fatal("DBOS initialization failed")
 	}
+	checkoutWF := dbos.RegisterWorkflow(dbosContext, checkoutWorkflow)
```

Comment:
Why do we need to provide a dbosContext when registering the workflow? I thought the context is only used when invoking the workflows.

Reply:
Good question. We might be OK with non-mockable Register functions.